app/jobs/application_job.rb

```diff
@@ -5,4 +5,10 @@ class ApplicationJob < ActiveJob::Base
 
   # Most jobs are safe to ignore if the underlying records are no longer available
   # discard_on ActiveJob::DeserializationError
+
+  def self.warning_error_classes(job)
+    job.instance_of?(RiscDeliveryJob) ?
+      RiscDeliveryJob::NETWORK_ERRORS + [RedisRateLimiter::LimitError]
+      : []
+  end
```

Comment:
I dislike this pattern of having (abstract) superclasses with knowledge of their concrete subclasses.

The advantage to having a method is that means it's an interface we can easily override in subclasses, and we can simplify the behavior too, no need to pass a `job` because we'd know the job from the class that's being called. That means the superclass here could have an empty/stub implementation:

```suggestion
  def self.warning_error_classes
    []
  end
```


And then in RiscDeliveryJob:

```
class RiscDeliveryJob < ApplicationJob
  # ...
  def self.warning_error_classes
    NETWORK_ERRORS + [ RedisRateLimiter::LimitError ]
  end
  # ...
end
```